src/requests/getForecast.js

```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+
+const getForecast = (
+  searchText,
+  setSelectedDate,
+  setForecasts,
+  setLocation,
+  setErrorMessage
+) => {
+  let endpoint = "https://cmd-shift-weather-app.onrender.com/forecast";
+
+  if (searchText) {
+    endpoint += `?city=${searchText}`;
+  }
+
+  return axios
+    .get(endpoint)
+    .then((response) => {
+      setSelectedDate(response.data.forecasts[0].date);
+      setForecasts(response.data.forecasts);
+      setLocation(response.data.location);
+    })
+    .catch((error) => {
```

Comment:
You have a bug here, if a user enters an invalid town then the results will never be displayed, even if the request is successful, as there is still an error message preventing causing the conditional render to stop it coming through:
```
      setLocation(response.data.location);
      setErrorMessage("");
    })
    .catch((error) => {
```